platforms/gke/base/core/workloads/cluster_credentials/versions.tf

```diff
@@ -0,0 +1,28 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+terraform {
+  required_version = ">= 1.5.7"
+
+  required_providers {
+    null = {
+      source  = "hashicorp/null"
+      version = "3.2.3"
+    }
```

Comment:
Instead of using `null_resource`s and the `null` provider, consider using the `terraform_data` resource, as recommended for Terraform >= 1.4.

https://developer.hashicorp.com/terraform/language/resources/terraform-data

Example: https://github.com/GoogleCloudPlatform/accelerated-platforms/blob/da6d3366c805064ad1bacb7a6e810d8375d16007/platforms/gke/base/use-cases/federated-learning/terraform/config_management/main.tf#L78